Set Obsoletes: for distro-provided package names

mjmac opened the pull request:
May improve dependency resolution.

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

mjmac committed:
Set Obsoletes: for distro-provided package names

May improve dependency resolution.

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

mjmac committed:
use pipeline test branch

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

brianjmurrell commented on Jenkinsfile:
```diff
@@ -39,5 +39,6 @@
 //@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@bmurrell/package-building-test-daos-matrix")
 
 packageBuildingPipelineDAOS(['distros': ['centos7', 'el8', 'leap15']])
```
```suggestion
packageBuildingPipelineDAOS(['distros': ['centos7', 'el8', 'leap15'],
                             'rpmlint_rpms_allow_errors': true])
```

brianjmurrell committed:
Switch pipeline-lib branches

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell commented:
I don't get it.  I don't see how [this result](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fgo-build-toolchain/detail/PR-5/16/pipeline/) is any different than [this result](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fmercury/detail/PR-77/56/pipeline/).  They both have stages which have `rpmlint`ing errors converted to warnings by `                             'rpmlint_rpms_allow_errors': true`.

Yet [the other] one allows the other stages to be `SUCCESS`ful and this one makes one of the stages `UNSTABLE` as a result of a different stage being `UNSTABLE`.

brianjmurrell committed:
Use the big a hammer

Just skip rpmlinting RPMs altogether as we cannot get Jenkins to work
properly with some UNSTABLE stages.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Remove pipeline-lib branch reference

The PR for it has landed to master.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell commented:
@mjmac This passes with pipeline-lib@master now.  Handing it back over to you.

mjmac commented:
Not necessary. This was an experiment to see if it would solve the problem ultimately solved by disabling the go-toolkit module in dnf.
